src/main/downloadsManager.ts

```diff
@@ -374,6 +378,46 @@ export class DownloadsManager extends JsonFileManager<DownloadedItems> {
         this.saveAll(downloads);
     };
 
+    private handleSelectDownload = (event: IpcMainInvokeEvent, startFrom: string) => {
+        return this.selectDefaultDownloadDirectory(
+            startFrom,
+            localizeMessage('main.downloadsManager.specifyDownloadsFolder', 'Specify the folder where files will download'),
+        );
+    }
+
+    private selectDefaultDownloadDirectory = async (startFrom: string, message: string) => {
+        log.debug('handleSelectDownload', startFrom);
+
+        const result = await dialog.showOpenDialog({defaultPath: startFrom || Config.downloadLocation,
+            message,
+            properties:
+        ['openDirectory', 'createDirectory', 'dontAddToRecent', 'promptToCreate']});
+        return result.filePaths[0];
+    }
+
+    private verifyMacAppStoreDownloadFolder = async (fileName: string) => {
+        let downloadLocation = Config.downloadLocation;
+
+        // eslint-disable-next-line no-undef
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        if (__IS_MAC_APP_STORE__ && downloadLocation) {
```

Comment:
If this is because of `__IS_MAC_APP_STORE__`, you might want to add a `.d.ts` somewhere that defines any of the globals you're using to avoid having to add these comments

Reply:
Yeah this seems like a good cleanup item to do in a future PR.